kafka/consumer/group.py

```diff
@@ -567,18 +564,34 @@ def topics(self):
         future = cluster.request_update()
         self._client.poll(future=future)
         cluster.need_all_topic_metadata = stash
+
+    def topics(self):
+        """Get all topics the user is authorized to view.
+
+        Returns:
+            set: topics
+        """
+        cluster = self._client.cluster
+        self._fetch_all_topic_metadata()
         return cluster.topics()
 
     def partitions_for_topic(self, topic):
         """Get metadata about the partitions for a given topic.
+        This method will issue a remote call to the server if it does not already
+        have any metadata about the given topic.
 
         Arguments:
             topic (str): Topic to check.
 
         Returns:
             set: Partition ids
         """
-        return self._client.cluster.partitions_for_topic(topic)
+        cluster = self._client.cluster
+        partitions = cluster.partitions_for_topic(topic)
+        if not partitions:
```

Comment:
The return value of `partitions` is `None`, so let's explicitly check for that: `if partitions is None`

Additionally, I noticed that a non-existent topic will also return `None`, so this will issue a remote call if the topic simply does not exist... I can't think of any perf problems this will cause and it's  already mentioned in my suggested docstring so I think it's fine.

```
In [1]: from kafka import KafkaConsumer

In [2]: kc = KafkaConsumer('test_topic', group_id='testings', bootstrap_servers=['172.28.0.3'])

In [3]: cluster = kc._client.cluster

In [4]: cluster.partitions_for_topic('test_poll_group_by_topic_true')

In [5]: cluster.partitions_for_topic('non_existent_topic')

In [6]: kc.topics()
Out[6]: {u'test_poll_group_by_topic_true'}

In [7]: cluster.partitions_for_topic('non_existent_topic')

In [8]: cluster.partitions_for_topic('test_poll_group_by_topic_true')
Out[8]: {0}
```